### DISCUSSION TITLE
Engine options for Quarto

### DISCUSSION THREAD

**Discussion by jleclai2** [quarto-dev/quarto-cli#10734] [Q&A] [answered]
### Description

I have a question about the engine options for Quarto.  When I create a new quarto document, the options for engine are 'knitr', 'jupyter', and 'none'.  I understand that if I'm using R I should use knitr, and jupyter when using python, but what is the none option for?  Thank you!

**Comment by mcanouil** [comment]
When you don't want code cell to be evaluated at all, i.e., no engine is started at all.

**Reply by jleclai2** [reply]
Thanks for your help.  Sorry, can you please clarify?  When I make a dummy .qmd file, and select engine = none, and add an R code chunk with the expression 1+1, it is evaluated and presents 2.  Can you please explain?  Thanks!

**Reply by mcanouil** [reply]
See @cderv's comment, I did not realise it was not a valid option.

By the way, I looked at the codebase and `julia` is not suggested as well as `markdown`.

- https://github.com/quarto-dev/quarto-cli/blob/7a0131d23e5525629de83959cd2d585f28603a92/src/resources/editor/tools/yaml/web-worker.js#L13877-L13889

````qmd
---
title: "Quarto Playground"
format: html
engine: markdown
---

```{r}
1 + 1
```
````

Won't have code cell executed.

**Comment by cderv** [comment]
Where do you see this `none` exactly ? 

Here is our doc about engine binding if you want to know how it works
https://quarto.org/docs/computations/execution-options.html#engine-binding

**Reply by jleclai2** [reply]
Thanks for your help - 

File -> New Quarto Document -> Engine (options in drop down menu include Knitr, Jupyter, and None).

When would we select "none"?  I am still able to get R code to execute when I select engine = none setting up my qmd file.

**Reply by mcanouil** [reply]
> File -> New Quarto Document -> Engine (options in drop down menu include Knitr, Jupyter, and None).

Where is that? Rstudio, VSCode, Positron? I am assuming the first, but not entirely sure.
The editor interface is independent of Quarto CLI. The integration is done by the editor (RStudio, etc.)

**Reply by jleclai2** [reply]
Sorry I should've clarified, RStudio!

**Reply by cderv** [reply, marked as answer]
In Rstudio IDE context, `(None)` means do not set anything in the YAML header. Then the default Engine binding behavior from Quarto applies. Which is described in the link I shared above. 

>  When I make a dummy .qmd file, and select engine = none, and add an R code chunk with the expression 1+1, it is evaluated and presents 2. Can you please explain? 

Reading this part of the doc should anwser your question. If you have a R cells in a document, then quarto automatically bind to the knitr engine if nothing else is specified.

**Reply by mcanouil** [reply]
So "None" is for "don't set an engine" while the other values are for "set the engine to XXX".
A bit confusing in my opinion.

**Reply by jleclai2** [reply]
Got it thank you both!

**Reply by cderv** [reply]
> So "None" is for "don't set an engine" while the other values are for "set the engine to XXX".
A bit confusing in my opinion.

Feel free to give feedback to IDE team and suggest another name for it: https://github.com/rstudio/rstudio
